src/Concordium/Client/Runner.hs

```diff
@@ -1057,61 +1052,62 @@ getEncryptedAmountTransferData senderAddr ettReceiver ettAmount idx secretKey =
-getDelegatorCooldown bs = do
-  Queries.bsWithUpdates bs $ \spv ups ->
-    case Types.chainParametersVersionFor spv of
-        Types.SCPV0 -> do
-          return Nothing
-        Types.SCPV1 -> do
-          currTime <- liftIO getCurrentTime
-          let cooldownTime = fromIntegral . Types.durationSeconds $ ups ^. Types.currentParameters . cpCooldownParameters . cpDelegatorCooldown
-          return $ Just $ addUTCTime cooldownTime currTime
+getDelegatorCooldown :: Queries.EChainParametersAndKeys -> IO (Maybe UTCTime)
+getDelegatorCooldown (Queries.EChainParametersAndKeys (ecpParams :: ChainParameters' cpv) _) = do
+  case Types.chainParametersVersion @cpv of
+      Types.SCPV0 -> do
+        return Nothing
+      Types.SCPV1 -> do
+        currTime <- liftIO getCurrentTime
+        let cooldownTime = fromIntegral . Types.durationSeconds $ ecpParams ^. cpCooldownParameters . cpDelegatorCooldown
+        return $ Just $ addUTCTime cooldownTime currTime
 
 -- |Query the chain for the given account. Fail if either the chain cannot be reached, or
 -- if the account does not exist.
-getAccountInfoOrDie :: (MonadIO m) => ID.AccountAddress -> BlockHashInput -> ClientMonad m Types.AccountInfo
-getAccountInfoOrDie senderAddr bhInput = do
-  res <- getAccountInfoV2 (Types.AccAddress senderAddr) bhInput
+getAccountInfoOrDie :: (MonadIO m) => Types.AccountIdentifier -> BlockHashInput -> ClientMonad m Types.AccountInfo
+getAccountInfoOrDie sender bhInput = do
+  res <- getAccountInfoV2 sender bhInput
   case res of
     StatusOk resp -> case grpcResponseVal resp of
       Left err -> logFatal ["Cannot decode account info response from the node: " <> err]
       Right v -> return v
-    StatusNotOk (NOT_FOUND, _) -> logFatal [[i|Account #{senderAddr} does not exist on the chain.|]] 
+    StatusNotOk (NOT_FOUND, _) -> logFatal [[i|No account with #{showAccountIdentifier sender} exists on the chain.|]] 
     StatusNotOk (status, err) -> logFatal [[i|GRPC response with status '#{status}': #{err}|]] 
     StatusInvalid -> logFatal ["GRPC response contained an invalid status code."]
-    RequestFailed err -> logFatal [[i|I/O error: #{err}|]]
+    RequestFailed err -> logFatal ["I/O error: " <> err]
 
 -- |Query the chain for the given pool. Fail if either the chain cannot be reached, or
 -- if the baker pool does not exist.
 getPoolStatusOrDie :: Maybe Types.BakerId ->  ClientMonad IO Queries.PoolStatus
 getPoolStatusOrDie mbid = do
-  let (bid, passiveDelegation) = case mbid of
-        Nothing -> (0, True)
-        Just bakerId -> (bakerId, False)
-  infoValue <- logFatalOnError =<< withBestBlockHash Nothing (getPoolStatus bid passiveDelegation)
-  case AE.fromJSON $ grpcResponseVal infoValue of
-    AE.Error err -> logFatal ["Cannot decode pool status response from the node: " ++ err]
-    AE.Success Nothing -> logFatal ["Could not query pool status from the chain."]
-    AE.Success (Just ps) -> return ps
+  psRes <- getPoolInfoV2 Best (fromMaybe 0 mbid)
```

Comment:
I don't think this is correct. If `Nothing` is supplied you are meant to query passive delegation if I recall correctly.

Reply:
Thanks for pointing that out. I will have to look into that a bit further.